doc/source/reference/arrays.dtypes.rst

```diff
@@ -296,7 +296,7 @@ String with comma-separated fields
       >>> dt = np.dtype("a3, 3u8, (3,4)a10")
 
 Type strings
-   Any string in :obj:`numpy.sctypeDict`.keys():
+   Any string in ``numpy.sctypeDict.keys()``:
```

Comment:
sctypeDict will be removed for NumPy2.0. Do we have a replacement?

Reply:
Can we add a warning to tell users sctypeDict will be removed for NumPy2.0 now? Then we can update this part for replacement if we have got a good replacement.

Reply:
@rgommers, @seberg, @ngoldbaum: what is the canonical list of strings that can be used for dtypes in NumPy2.0, once `numpy.sctypeDict` disappears?

Reply:
For the list of canonical Python names for dtypes, see gh-24651. There should not be a new list of strings accessible in the API. The examples right below this line are already fine, so I recommend replace this line with _"Any string name of a NumPy dtype, e.g.:"_

Reply:
I updated the commit based on @rgommers suggestion.